tests/Makefile

```diff
@@ -2,7 +2,7 @@
 # To compile on SunOS: add "-lsocket -lnsl" to LDFLAGS
 # To compile with PKCS11: add "-lpkcs11-helper" to LDFLAGS
 
-CFLAGS	?= -O2
+CFLAGS	?= -O0
```

Comment:
Mainly as a note to myself....

I don't think this is a good way to suppress a warning.

The actual warning (which is gcc specific) is `-Wclobbered`, so should be suppressed with `-Wno-clobbered`. However, clang doesn't recognise the parameter `-Wno-clobbered` and we don't have a compiler specific means of passing CFLAGS at the moment, so that won't work.

Therefore, if we needed to suppress the warning, may be a better means to suppress it would be in the source file itself to detect gcc and `pragma` a warning away. The following seems to work:
```
 #if defined(MBEDTLS_CHECK_PARAMS)
#if defined(__GNUC__) && !defined(__clang__)
#pragma GCC diagnostic ignored "-Wno-uninitialized"
#endif

+#endif /* MBEDTLS_CHECK_PARAMS */
```

However - I don't think we should be disabling warnings here, so this shouldn't be needed.